packages/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp

```diff
@@ -869,32 +883,57 @@ void ReanimatedModuleProxy::performOperations() {
     jsPropsUpdater.call(rt, viewTag, nonAnimatableProps);
   }
 
-  bool hasLayoutUpdates = false;
+  HasLayoutupdates hasLayoutUpdates;
 #ifdef ANDROID
   bool hasPropsToRevert = updatesRegistryManager_->hasPropsToRevert();
 #else
   bool hasPropsToRevert = false;
 #endif
 
   if (!hasPropsToRevert) {
+    for (const auto &[shadowNode, props] : transitionUpdatesBatch) {
+      if (isThereAnyLayoutProp(props)) {
+        hasLayoutUpdates.cssTransition = true;
+        break;
+      }
+    }
     for (const auto &[shadowNode, props] : updatesBatch) {
       if (isThereAnyLayoutProp(rt, props->asObject(rt))) {
-        hasLayoutUpdates = true;
+        hasLayoutUpdates.workletAnimation = true;
+        break;
+      }
+    }
+    for (const auto &[shadowNode, props] : animationUpdatesBatch) {
+      if (isThereAnyLayoutProp(props)) {
+        hasLayoutUpdates.cssAnimation = true;
         break;
       }
     }
   }
 
-  if (!hasLayoutUpdates && !hasPropsToRevert) {
-    // If there's no layout props to be updated, we can apply the updates
-    // directly onto the components and skip the commit.
+  // If there's no layout props to be updated, we can apply the updates
+  // directly onto the components and skip the commit.
+  if (!hasPropsToRevert && hasLayoutUpdates.hasAny()) {
     ReanimatedSystraceSection s(
         "ReanimatedModuleProxy::synchronouslyUpdateUIProps");
-    for (const auto &[shadowNode, props] : updatesBatch) {
-      Tag tag = shadowNode->getTag();
-      synchronouslyUpdateUIPropsFunction_(rt, tag, props->asObject(rt));
+    if (!hasLayoutUpdates.cssTransition) {
+      for (const auto &[shadowNode, props] : transitionUpdatesBatch) {
+        Tag tag = shadowNode->getTag();
+        synchronouslyUpdateUIPropsByDynamicFunction_(tag, props);
```

Comment:
Code related to `synchronouslyUpdateUIProps` will be removed after the merge of https://github.com/software-mansion/react-native-reanimated/pull/7014